### PR TITLE
[Printer] Try extends PHPStan Printer

### PR DESCRIPTION
**Pull request by samsonasik**
Per https://github.com/rectorphp/rector-src/pull/6517#issuecomment-2503825512

### CODE REVIEW

File: src/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php [hunk 1]
```diff
@@ -295,7 +294,6 @@ public function processNodes(
         $this->nodeScopeResolverProcessNodes($stmts, $scope, $nodeCallback);
 
         $nodeTraverser = new NodeTraverser();
-        $nodeTraverser->addVisitor(new WrappedNodeRestoringNodeVisitor());
```
Comment (samsonasik) [inline src/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:298]: @ondrejmirtes I tried remove this unwrap, and it somehow cause invalid print:

```diff
Failed on fixture file "mirror_assign_comment.php.inc"
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 {
     public static function run()
     {
-        $response = match ($_SERVER['REQUEST_METHOD']) {
+        $response = match (__phpstanRembered($_SERVER['REQUEST_METHOD'])) {
```

see "__phpstanRembered" printered above, and cause error:

```
2) Rector\Tests\CodeQuality\Rector\ClassMethod\OptionalParametersAfterRequiredRector\OptionalParametersAfterRequiredRectorTest::test with data set #7 ('/Users/samsonasik/www/rector-...hp.inc')
assert($subStartPos >= 0 && $subEndPos >= 0)

/home/runner/work/rector-src/rector-src/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinterAbstract.php:688
/home/runner/work/rector-src/rector-src/src/PhpParser/Printer/BetterStandardPrinter.php:116
```

**Comment by ondrejmirtes** [comment]
Alright, I get it. Please forget what I said, you need to unwrap the expression by yourself and the PHPStan Printer class isn't usable for your use-case.

File: src/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php (hunk 1 above)
Comment (samsonasik) [reply src/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php:298]: @ondrejmirtes it seems `__phpstanRembered` string make wrapping node print, and cause above output

https://github.com/phpstan/phpstan-src/blob/1abeeb77771fc3fece5c7196f75dfc1478840515/src/Node/Printer/Printer.php#L80

**Comment by samsonasik** [comment]
Ok, let's close then ;)

**Pull request closed without merging**